### PR TITLE
Multilingual Feature -> Update Translations

### PR DESCRIPTION
**Pull request by Ali-Salman29**

### CODE REVIEW

File: openedx/features/wikimedia_features/meta_translations/models.py [hunk 1]
```diff
@@ -137,7 +137,7 @@ class WikiTranslation(models.Model):
     """
     Store translations fetched from wiki learn meta, will also manage mapping of source and target blocks.
     """
-    target_block = models.ForeignKey(CourseBlock, on_delete=models.CASCADE)
+    target_block = models.ForeignKey(CourseBlock, on_delete=models.CASCADE, related_name="course_block")
```
Comment (tehreem-sadat) [inline openedx/features/wikimedia_features/meta_translations/models.py:140]: why do we need this ?

File: openedx/features/wikimedia_features/meta_translations/api/v0/views.py [hunk 2]
```diff
@@ -222,3 +250,15 @@ def get(self, request, *args, **kwargs):
         data = [self._course_version_format(key) for key in base_course_keys]
         json_data = dict(data)
         return Response(json_data, status=status.HTTP_200_OK)
+
+class CourseBlockViewSet(viewsets.ModelViewSet):
+    """
+    Viewset to update Couse Block and Wiki Translations together
```
Comment (tehreem-sadat) [inline openedx/features/wikimedia_features/meta_translations/api/v0/views.py:256]: Add sample API(s) in docstring.

File: openedx/features/wikimedia_features/meta_translations/api/v0/serializers.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        """
+        Get user from request
+        """
+        request = self.context.get('request', None)
+        if request:
+            return request.user
+    
+    def _validate_translations(self, instances, validated_data):
+        """
+        A custom validation that checks wiki_traslations and course block instences
```
Comment (tehreem-sadat) [inline openedx/features/wikimedia_features/meta_translations/api/v0/serializers.py:62]:
```suggestion
        A custom validation that checks wiki_translations and course block instances.
```

File: openedx/features/wikimedia_features/meta_translations/api/v0/serializers.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        wiki_translations must include all the instences of block
```
Comment (tehreem-sadat) [inline openedx/features/wikimedia_features/meta_translations/api/v0/serializers.py:63]:
```suggestion
        wiki_translations must include all the instances of block
```

File: openedx/features/wikimedia_features/meta_translations/api/v0/serializers.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            wiki_translations = WikiTranslation.objects.filter(target_block=instance)
+            if 'wiki_translations' in validated_data:
+                block_translations = validated_data.pop('wiki_translations')
+                self._validate_translations(wiki_translations, block_translations)
+                for block_translation in block_translations:
+                    id = block_translation.pop('id')
+                    block_translation['approved_by'] = user
+                    wiki_translations.filter(id=id).update(**block_translation)
+            else:
+                wiki_translations.update(applied=applied, approved_by=user)
```
Comment (tehreem-sadat) [inline openedx/features/wikimedia_features/meta_translations/api/v0/serializers.py:99]: NIT: please verify that calling update is not disturbing our post_save signal.

**Review by tehreem-sadat — APPROVED** [review]

File: openedx/features/wikimedia_features/meta_translations/models.py (hunk 1 above)
Comment (Ali-Salman29) [reply openedx/features/wikimedia_features/meta_translations/models.py:140]: There are multiple ForeginKeys in the Course Block, that's why the _set command was not working in a serializer. To get wiki_translations from a block we need to name it.
The default name was not working i.e wiki_translation_set

File: openedx/features/wikimedia_features/meta_translations/signals.py [hunk 6]
```diff
@@ -14,9 +14,9 @@ def add_language_and_set_mapping_update(sender, instance, created, **kwargs):
     On every new Mapping Object (WikiTranslation), set update_mapping to True and update languages so that
     next send_translation cron job will sync updated blocks data to wikimedia Meta for translations.
     """
+    import pdb; pdb.set_trace();
```
Comment (tehreem-sadat) [inline openedx/features/wikimedia_features/meta_translations/signals.py:17]: remove pdb.

**Review by tehreem-sadat — APPROVED** [review]

**Pull request merged**